From c85004082a4c451de20e80349ff3f5cf987d1dab Mon Sep 17 00:00:00 2001
From: Spezialk-dev <spezcialK@protonmail.com>
Date: Fri, 10 Jan 2025 11:01:55 +0100
Subject: [PATCH] fixed kernel update to 6.4 with breaking api change

---
 lkm.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/lkm.c b/lkm.c
index 98a573eaa51ce58cf0502c0683ee1f4e565bdc84..75e95cd0b8dfa369f0e57fed894b4c08eaa11448 100644
--- a/lkm.c
+++ b/lkm.c
@@ -376,9 +376,15 @@ static int __init chr_dev_init(void)
 {
 	int i;
 	if (register_chrdev(FMEM_MAJOR,"fmem",&memory_fops))
-		printk("unable to get major %d for memory devs\n", FMEM_MAJOR);
+	  printk("unable to get major %d for memory devs\n", FMEM_MAJOR);
 
+	// fixing a problem that is caused by Kernels newer than 6.4 since since then the class_create only expects on parameter
+	#if(LINUX_VERSION_CODE >=KERNEL_VERSION(6, 4, 0))
+	mem_class = class_create("fmem");
+	#else
 	mem_class = class_create(THIS_MODULE, "fmem");
+	#endif
+	
 	for (i = 0; i < ARRAY_SIZE(devlist); i++) {
 		device_create(mem_class, NULL, MKDEV(FMEM_MAJOR, devlist[i].minor), NULL, devlist[i].name);
 	}
